app/src/main/java/com/keylesspalace/tusky/components/timeline/TimelineFragment.kt

```diff
@@ -201,12 +234,39 @@ class TimelineFragment :
                 if (positionStart == 0 && adapter.itemCount != itemCount) {
                     binding.recyclerView.post {
                         if (getView() != null) {
-                            if (isSwipeToRefreshEnabled) {
+                            // Swiping and prefer oldest-first? Stay at the same position.
+                            // Otherwise jump to the top.
+                            if (isSwipeToRefreshEnabled && readingOrder == ReadingOrder.OLDEST_FIRST) {
                                 binding.recyclerView.scrollBy(0, Utils.dpToPx(requireContext(), -30))
                             } else binding.recyclerView.scrollToPosition(0)
                         }
                     }
                 }
+
+                // If this insert was because the user clicked "Load more" then ensure the status
```

Comment:
Regarding comment / readability: all of this could move to a private method; maybe `ensureLastStatusVisible()`.

And there only the comment about the bug needs to remain.